File: Algorithms/Greedy/MinSquares.java

public class MinSquares {

    public static int getMinSquares(int length, int width) {

        int result = 0;

        int largeSide = Math.max(length, width),
            smallSide = Math.min(length, width);

        while (smallSide > 0) {

            result += largeSide / smallSide;

            int temp = largeSide % smallSide;

            largeSide = smallSide;
            smallSide = temp;

        }

        return result;

    }

    public static void main(String[] args) {

        System.out.println(getMinSquares(13, 29));

    }

}

File: Algorithms/Greedy/MinSquares.js
"use strict";

let getMinSquares = (length, width) => {

    let result = 0;

    let largeSide = Math.max(length, width),
        smallSide = Math.min(length, width);

    while (smallSide > 0) {

        result += Math.floor(largeSide / smallSide);

        let temp = largeSide % smallSide;

        largeSide = smallSide;
        smallSide = temp;

    }

    return result;

}

console.log(getMinSquares(13, 29));
